File: forge-ui/src/lib/ollamaExport.ts
/** Build Ollama import commands from completed export job output paths. */
export function buildOllamaCommandsFromExport(
  outputPathsJson: string | undefined,
  modelName: string,
): string[] {
  if (!outputPathsJson) return [];
  let outputs: Record<string, string>;
  try {
    outputs = JSON.parse(outputPathsJson) as Record<string, string>;
  } catch {
    return [];
  }

  const safeName = modelName.trim().replace(/\s+/g, "-") || "seiso-model";
  const ggufEntries = Object.entries(outputs).filter(([key]) => key.toLowerCase().includes("gguf"));
  const commands: string[] = [];

  for (const [key, rawPath] of ggufEntries) {
    const dir = rawPath.replace(/[/\\][^/\\]+$/, "");
    const quant = key.startsWith("gguf_") ? key.slice("gguf_".length) : key;
    const name = ggufEntries.length === 1 ? safeName : `${safeName}-${quant}`;
    commands.push(`ollama create ${name} -f ${dir}/Modelfile`);
  }

  return commands;
}

export function extractOllamaCommandsFromLogs(logs: string[]): string[] {
  return logs
    .filter((line) => line.startsWith("Ollama: "))
    .map((line) => line.slice("Ollama: ".length).trim())
    .filter(Boolean);
}

File: forge/services/ollama_export.py
"""Build Ollama import commands from Seiso export outputs."""

from __future__ import annotations

from pathlib import Path


def build_ollama_create_commands(
    outputs: dict[str, str],
    *,
    model_name: str,
) -> list[str]:
    """Return shell commands to import GGUF export artifacts into Ollama."""
    commands: list[str] = []
    safe_name = model_name.strip().replace(" ", "-") or "seiso-model"

    for key, raw_path in sorted(outputs.items()):
        if "gguf" not in key.lower():
            continue
        gguf_path = Path(raw_path)
        modelfile = gguf_path.parent / "Modelfile"
        if not modelfile.is_file():
            continue
        quant = key.removeprefix("gguf_") if key.startswith("gguf_") else key
        name = safe_name if len(outputs) == 1 else f"{safe_name}-{quant}"
        commands.append(f"ollama create {name} -f {modelfile}")

    return commands
